packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -133,30 +188,75 @@ export class Arn {
    *
    * @returns an ArnComponents object which allows access to the various
    *      components of the ARN.
+   *
+   * @deprecated use split instead
    */
   public static parse(arn: string, sepIfToken: string = '/', hasName: boolean = true): ArnComponents {
+    let arnFormat: ArnFormat;
+    if (!hasName) {
+      arnFormat = ArnFormat.NO_RESOURCE_NAME;
+    } else {
+      arnFormat = sepIfToken === '/' ? ArnFormat.SLASH_RESOURCE_NAME : ArnFormat.COLON_RESOURCE_NAME;
+    }
+    return this.split(arn, arnFormat);
+  }
+
+  /**
+   * Splits the provided ARN into its components.
+   * Works both if 'arn' is a string like 'arn:aws:s3:::bucket',
+   * and a Token representing a dynamic CloudFormation expression
+   * (in which case the returned components will also be dynamic CloudFormation expressions,
+   * encoded as Tokens).
+   *
+   * @param arn the ARN to split into its components
+   * @param arnFormat the expected format of 'arn' - depends on what format the service 'arn' represents uses
+   */
+  public static split(arn: string, arnFormat: ArnFormat): ArnComponents {
```

Comment:
Ah well poops. I agree with the decision to make this a separate function but I'm not too fond of the name `split`... yet, I also don't have a better one. What to do when the prime real estate name is already taken... 🙄

(I guess there is at least some sort of parallel with `path.split()` ...)